### DISCUSSION TITLE
How can I set the window icon?

### DISCUSSION THREAD

**Discussion by Jak2k** [AparokshaUI/Adwaita#51] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/93d84673-4901-4abe-953d-8ed5e91b3874)

**Comment by david-swift** [comment, marked as answer]
# In the shell

![Screenshot From 2024-09-08 08-35-37](https://github.com/user-attachments/assets/79f018e2-89eb-4308-a172-40b299e00a8c)

If you run the app in a Flatpak (e.g. based on the [template](https://github.com/AparokshaUI/AdwaitaTemplate)), follow the steps below. In case you simply want to try to build the template and show the icon (no custom icon), start at step 4.

1. [Design the icon based on the SVG template.](https://blogs.gnome.org/tbernard/2019/12/30/designing-an-icon-for-your-app/)
2. Save the icons under `./data/icons` (or any other directory in your project). Normally, you would follow the following naming scheme (`<app-id> should be replaced by the [reverse DNS style identifier](https://developer.gnome.org/documentation/tutorials/application-id.html):
- `<app-id>.Source.svg` for the file based on the SVG template.
- `<app-id>.svg` for the icon visible in the dash and app overview grid
- `<app-id>-symbolic.svg` for the symbolic icon which is visible for notifications for example
3. In the Flatpak manifest (`./<app-id>.json`), under `build-commands`, install the "regular" and the symbolic app icons:
```json
"install -Dm644 data/icons/<app-id>.svg $DESTDIR/app/share/icons/hicolor/scalable/apps/<app-id>.svg",
"install -Dm644 data/icons/<app-id>-symbolic.svg $DESTDIR/app/share/icons/hicolor/symbolic/apps/<app-id>-symbolic.svg"
```
4. If you run the app using the GNOME Builder's or the VS Code Flatpak extension's run command, those installation steps will not be executed and the app icon will not be visible. Instead:
- In GNOME Builder, open the menu next to the hammer icon and click `Export`. Open the Flatpak file in GNOME Software (the icon won't be visible) and install. Now, in the app overview grid or when running the installed app, you should see the icon. If you run using GNOME Builder's run command, as the icon is saved for the app's id, your app will have the correct icon.
- The VS Code Extension doesn't seem to have installation functionality, so install via the [Flatpak Builder command](https://docs.flatpak.org/en/latest/first-build.html). After the installation, if you run using the VS Code Extension's run command, your app will show the correct icon.


# Access in the app's UI

![Screenshot From 2024-09-08 08-40-51](https://github.com/user-attachments/assets/9b67ac5e-d712-428e-a4ae-1d06dbe3949f)

If you want to display the icon in the app's UI using [`Icon`](https://aparokshaui.github.io/adwaita-swift/documentation/adwaita/icon), for example in an [`aboutDialog`](https://aparokshaui.github.io/adwaita-swift/documentation/adwaita/view/aboutdialog(visible:app:developer:version:icon:website:issues:)), use `Icon.custom(name: "<app-id")`. The about dialog above uses the following code:

```swift
.aboutDialog(
    visible: $about,
    app: "AdwaitaTemplate",
    developer: "david-swift",
    version: "dev",
    icon: .custom(name: "io.github.AparokshaUI.AdwaitaTemplate"), // The app icon
    website: .init(string: "https://github.com/AparokshaUI/AdwaitaTemplate")!,
    issues: .init(string: "https://github.com/AparokshaUI/AdwaitaTemplate/issues")!
)
```

**Reply by Jak2k** [reply]
I changed nothing of the code, but now it works. Still, thank you for the response.
